csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/Assets.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            // If there are more than one framework we need to pick just one.
+            // For now, we just pick the first one.
+            var references = json
+                .GetProperty("targets")?
+                .GetFirstPropertyValue()?
+                .ToObject<Dictionary<string, ReferenceInfo>>();
+
+            if (references is null)
+            {
+                progressMonitor.LogDebug("No references found in the targets section in the assets file.");
+                return Array.Empty<string>();
+            }
+
+            // Find all the compile dependencies for each reference and
+            // create the relative path to the dependency.
+            var dependencies = references
+                .SelectMany(r =>
+                {
+                    var info = r.Value;
+                    var name = r.Key.ToLowerInvariant();
+                    if (info.Type != "package")
+                    {
+                        return Array.Empty<string>();
+                    }
+
+                    // If this is a .NET framework reference then include everything.
+                    return netFrameworks.Any(framework => name.StartsWith(framework))
+                        ? new[] { name }
+                        : info
+                            .Compile
+                            .Select(p => Path.Combine(name, ParseFilePath(p.Key)));
+                })
+                .ToList();
+
+            return dependencies;
+        }
+
+        /// <summary>
+        /// Gets the framework dependencies from the assets file.
+        ///
+        /// Example:
+        /// "project": {
+        //     "version": "1.0.0",
+        //     "frameworks": {
+        //         "net7.0": {
+        //             "frameworkReferences": {
+        //                 "Microsoft.AspNetCore.App": {
+        //                     "privateAssets": "none"
+        //                 },
+        //                 "Microsoft.NETCore.App": {
+        //                     "privateAssets": "all"
+        //                 }
+        //             }
+        //         }
+        //     }
+        // }
+        //
+        /// Returns { "microsoft.aspnetcore.app.ref", "microsoft.netcore.app.ref" }
+        /// </summary>
+        private IEnumerable<string> GetFrameworkDependencies(JObject json)
+        {
+
+            var frameworks = json
+                .GetProperty("project")?
+                .GetProperty("frameworks");
+
+            if (frameworks is null)
+            {
+                progressMonitor.LogDebug("No framework section in assets.json.");
+                return Array.Empty<string>();
+            }
+
+            // If there are more than one framework we need to pick just one.
+            // For now, we just pick the first one.
+            var references = frameworks
+                .GetFirstPropertyValue()?
+                .GetProperty("frameworkReferences");
```

Comment:
We might get better results if we chose all. There's a high chance that .net framework variants use the same assembly names, so our deduplication logic would choose the latest assembly version (instead of choosing one randomly here).

Also, do we know if the order here is deterministic?

Reply:
Yes, that makes sense.
If it turns out that the de-duplication is not sufficient then we can narrow it again.